### ISSUE TITLE
Graphql autocomplete support 

### ISSUE THREAD

**Issue by collegeimprovements** [Huachao/vscode-restclient#575]
Once we have setup/defined the endpoint, it would be really nice to get autocomplete support for queries just like most graphiql clients. They do it with graphql-introspection.

**Comment by Frank5000** [comment]
Might be asking for too much here but a documentation explorer similar to GraphiQL or Insomnia would be perfect

**Comment by tvvignesh** [comment]
Yup. If this is available, I can move completely away from Insomnia to Restclient rather than switching in between the two. Autocomplete for variables, queries, mutations based on introspection from the GQL endpoint would be great.

**Comment by acao** [comment]
hey there! GraphQL LSP/'GraphiQL maintainer here. there are a number of ways you can approach this! 

reading your source a bit, I think your most ideal option would be to implement our [`graphql-language-service`](https://github.com/graphql/graphiql/tree/main/packages/graphql-language-service/src/interface) direclty [here](https://github.com/Huachao/vscode-restclient/blob/master/src/providers/requestVariableCompletionItemProvider.ts), and in other applicable providers.

and yes, you can use `graphql-introspection` to request the schema to pass to `getAutocompleteSuggestions`, etc, but implementing this with fetch/etc & `graphql-js` is easy enough as it is using:
1. getIntrospectionQuery() to build the introspection query. there are some nifty options here you can expose to users
2. `buildClientSchema()` (be sure to pass `<body>.data` and not just the raw parsed `<body>` to buildClientSchema(), I and everyone always makes that mistake 😆 ).

For comparison, Insomnia and Altair use `codemirror-graphql`, other tools use `monaco-graphql`, both of which are our libraries that implement these `graphql-language-service` methods (thus why vscode-graphql and graphiql are in the same monorepo)

The only gotcha may be the line offsets - you may need to add or subtract by one to get the right positions, because codemirror + monaco are 0 based line offsets, and vscode's line offsets start at 1, i think? or the opposite I can never remember 😆 so either it will work automatically or you'll need to fiddle with offsets for Range. 

This low level lib is not always as easy to use as the higher level service layers, but I think would be the best option. The main reason that our LSP server would be way too much here would be the custom parsing of your .http files, and the LSP server knows about SDL and can resolve references across the file tree, which adds 80% of the complexity of our server, and that would just get in the way of your much simpler needs

If you need any help, feel free to ping me on discord or tag me in a PR!

(to add, your choice to forego the LSP model for language feature support is to your benefit here, though it might not scale well for HUGE graphql queries, 10k+ lines formatted huge)
